index.html

```diff
@@ -160,15 +238,36 @@
 
     points.enter().append('circle')
       .attr('class', 'point')
-      .attr('r', 5);
+      .attr('r', 2);
 
     points
-      .attr('cx', function (d) { return x(d.unemployment); })
-      .attr('cy', function (d) { return y(d.inflation); });
+      .attr('cx', function (a) { return x(a.costs); })
+      .attr('cy', function (a) { return y(a.visits); });
 
     points.exit().remove();
 
     // HISTOGRAMS
+    var map = txData.map(function(d) { return d.costs });
+
+    var histogram = d3.layout.histogram()
+                      .bins(x.ticks())
+                      (map);
+
+    var bars = svg.selectAll(".bar")
+                  .data(histogram)
+                  .enter().append("g")
```

Comment:
Note that because we immediately call `.enter()` after joining the data, the `bars` variable contains just the enter selection. This means that once we have added all of the bars the first time `update()` is called, we'll never change them again — there's always the same number of bars, so nothing to enter, thus all of this code is called on an empty selection the second and subsequent times it runs. We would instead call `bars.enter()...` on a new line so that we can continue to access the full `bars` selection.

Compare and contrast to how `points` is done above.